src/main/java/mate/academy/library/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.library.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.library.dto.category.CategoryResponseDto;
+import mate.academy.library.dto.category.CategoryRequestDto;
+import mate.academy.library.exception.EntityNotFoundException;
+import mate.academy.library.mapper.CategoryMapper;
+import mate.academy.library.model.Category;
+import mate.academy.library.repository.category.CategoryRepository;
+import mate.academy.library.service.CategoryService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryResponseDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryResponseDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(

```

Comment:
or you can use 
return categoryRepository.findById(id).map(categoryMapper::toDto).orElseThrow(....